### PR TITLE
Add test for CASSANDRA-8072

### PR DESCRIPTION
**Pull request by beobal**
Verifies that a node which has been decommed and wiped can rejoin and,
by implication, that it can successfully establish gossip connections
with existing seeds.

### CODE REVIEW

File: bootstrap_test.py [hunk 1]
```diff
@@ -351,6 +351,41 @@ def decommissioned_wiped_node_can_join_test(self):
...
+        """
+        cluster = self.cluster
+        cluster.populate(1)
+        cluster.start(wait_for_binary_proto=True)
+
+        # Add a new node, bootstrap=True ensures that it is not a seed
+        node2 = new_node(cluster, bootstrap=True)
+        node2.start(wait_for_binary_proto=True, wait_other_notice=True)
+
+        # Decommision the new node and kill it
```
Comment (ptnapoleon) [inline bootstrap_test.py]: nit: `Decommission`

File: bootstrap_test.py [hunk 2]
```diff
@@ -351,6 +351,41 @@ def decommissioned_wiped_node_can_join_test(self):
...
+
+        commitlog_dir = os.path.join(node2.get_path(), 'commitlogs')
+        debug("Deleting {}".format(commitlog_dir))
+        shutil.rmtree(commitlog_dir)
+
+        # Now start it, it should be allowed to join
+        mark = node2.mark_log()
+        debug("Restarting wiped node2")
+        node2.start(wait_other_notice=False)
+        node2.watch_log_for("JOINING:", from_mark=mark)
```
Comment (ptnapoleon) [inline bootstrap_test.py]: Is this sufficient to assert it has joined appropriately?

**Comment by ptnapoleon** [comment]
Have you checked this fails before 8072, and passes with the fix?

File: bootstrap_test.py (hunk 2 above)
Comment (beobal) [inline bootstrap_test.py]: Yes, the crux of 8072 is that the initial shadow round of gossip fails in this scenario. This is called from SS::prepareToJoin and unless it completes successfully, the wiped node won't move on to actually join the ring.

File: bootstrap_test.py (hunk 1 above)
Comment (beobal) [inline bootstrap_test.py]: fixed, thanks

**Comment by beobal** [comment]
Yes. This is actually @stef1927 's test, just slightly reworked to take accommodate the CASSANDRA-6696 changes. Both she and I have verified that it fails without and passes with the patch.

**Comment by ptnapoleon** [comment]
+1

**Pull request merged**